kubernetes/samples/scripts/terraform/vcn.tf

```diff
@@ -0,0 +1,414 @@
+
+// Compartment in which to create the clsuter resources.
+variable "compartment_name" {}
+variable "compartment_ocid" {}
+
+variable vcn_cidr_prefix { default = "10.0" }
+variable "vcn_cidr" {
+  default = "10.0.0.0/16"
+}
+
+# ------------------------------------
+
+/*
+ * Create a VCN. 
+ * a DNS label with the name of the cluster is attached to the VCN.
+ * The creation of the vcn also creates the default route table, security list, and dhcp options.
+ */
+resource "oci_core_virtual_network" "oke-vcn" {
+  cidr_block     = "${var.vcn_cidr}"
+  dns_label      = "${var.cluster_name}vcn"
+  compartment_id = "${var.compartment_ocid}"
+  display_name   = "${var.cluster_name}_vcn"
+}
+
+/*
+ * An internet gateway is created in the relevant compartment attached to the created VCN. 
+ */
+resource "oci_core_internet_gateway" "oke-igateway" {
+  compartment_id = "${var.compartment_ocid}"
+  display_name   = "${var.cluster_name}-igateway"
+  vcn_id         = "${oci_core_virtual_network.oke-vcn.id}"
+}
+
+/*
+ * Configures the default route table that was created when the VCN was created.
+ * The default route is pointed to the internet gateway that was created. 
+ */
+
+resource "oci_core_default_route_table" "oke-default-route-table" {
+  manage_default_resource_id = "${oci_core_virtual_network.oke-vcn.default_route_table_id}"
+  display_name               = "${var.cluster_name}-default-route-table"
+
+  route_rules {
+    cidr_block        = "0.0.0.0/0"
```

Comment:
this is pretty open, you should make sure you explain this in the readme... so people know what they are getting